client/rest-high-level/roles.yml

```diff
@@ -5,6 +5,7 @@ admin:
     - names: '*'
       privileges:
         - all
+      allow_restricted_indices: true
```

Comment:
Is this so that we can delete `.watches` when it gets reinstated?

Similar question for the other instances of this - why do we need it? When we flip the switch to disallowing system index access via REST, can we remove these?

Reply:
I did this because I was getting security errors on the feature reset API calls on tests where security is enabled. I will take another look at the feature reset API code and see if there's an approach where we can call the feature reset API without needing the caller to allow restricted indices.

At the very least I will add a test somewhere that validates the behavior I saw to make it easier to discuss options.

Reply:
I've dug in on this and I don't really see a way around it. If security is enabled and you want to delete a restricted index, you need a role with `allow_restricted_indices: true`. So I'll just add the behavior-confirming test.

Reply:
Since we are no longer allowing restricted indices in test roles, I'm going to resolve this conversation.